github/integration_test.go

```diff
@@ -562,6 +562,81 @@ var _ = Describe("GitHub Provider", func() {
 
 	})
 
+	It("should be possible to download files from path and branch specified with nested directory and max options", func() {
+
+		userRepoRef := newUserRepoRef(testUser, testUserRepoName)
+
+		userRepo, err := c.UserRepositories().Get(ctx, userRepoRef)
+		Expect(err).ToNot(HaveOccurred())
+
+		defaultBranch := userRepo.Get().DefaultBranch
+
+		path0 := "clustersDir/cluster/machine.yaml"
+		content0 := "machine yaml content"
+		path1 := "clustersDir/cluster/machine1.yaml"
+		content1 := "machine1 yaml content"
+		path2 := "clustersDir/cluster2/clusterSubDir/machine2.yaml"
+		content2 := "machine2 yaml content"
+
+		files := []gitprovider.CommitFile{
+			{
+				Path:    &path0,
+				Content: &content0,
+			},
+			{
+				Path:    &path1,
+				Content: &content1,
+			},
+			{
+				Path:    &path2,
+				Content: &content2,
+			},
+		}
+
+		commitFiles := make([]gitprovider.CommitFile, 0)
+		for _, file := range files {
+			path := file.Path
+			content := file.Content
+			commitFiles = append(commitFiles, gitprovider.CommitFile{
+				Path:    path,
+				Content: content,
+			})
+		}
+
+		_, err = userRepo.Commits().Create(ctx, *defaultBranch, "added config files", commitFiles)
+		Expect(err).ToNot(HaveOccurred())
+
+		//Recursive option check
+		options := gitprovider.FilesGetOptions{Recursive: *gitprovider.BoolVar(true)}
+		downloadedFiles, err := userRepo.Files().Get(ctx, "clustersDir", *defaultBranch, &options)
+		Expect(err).ToNot(HaveOccurred())
+		for ind, downloadedFile := range downloadedFiles {
+			Expect(*downloadedFile).To(Equal(files[ind]))
+		}
+
+		//Max depth option check
+		//should be files in 2 levels: machine.yaml and machine1.yaml in level 2 (level 1 has cluster and cluster2 dirs)
+		options = gitprovider.FilesGetOptions{Recursive: *gitprovider.BoolVar(true), MaxDepth: 2}
```

Comment:
can we add 2 more tests:
- all files with MaxDepth 3
- 2 files with MaxDepth and MaxFiles 2